### PR TITLE
test: 💍 Add tests for MQL expression generator

### PR DESCRIPTION
**Pull request by ZedLi**
## Description
Added tests for generating MQL expressions. There was a small bug with all of the greater/less than operators when working with non strings. We don't use any of them yet but I changed how we sanitize. 

## Checklist:

- ~[ ] I have added before and after screenshots for UI changes~
- ~[ ] I have added JSON response output for API changes~
- ~[ ] I have added steps to reproduce and test for bug fixes in the description~
- ~[ ] I have commented on my code, particularly in hard-to-understand areas~
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: addons/api/tests/unit/utils/mql-query-test.js [hunk 1]
```diff
@@ -0,0 +1,208 @@
+import {
+  generateMQLSearchExpression,
+  generateMQLFilterExpression,
+  generateMQLExpression,
+} from 'api/utils/mql-query';
+import { module, test } from 'qunit';
+
+module('Unit | Utility | mql-query', function () {
+  module('generateMQLSearchExpression tests', function () {
```
Comment (ZedLi) [inline addons/api/tests/unit/utils/mql-query-test.js:9]: I started using nested modules 🤷, I like the organization more

Comment (cameronperera) [reply addons/api/tests/unit/utils/mql-query-test.js:9]: I love it. @calcaide has been looking into this so he may have some input. However, I like the clear use of it in a pretty simple test file.

**Review by cameronperera — APPROVED** [review]
LGTM

File: addons/api/tests/unit/utils/mql-query-test.js (hunk 1 above)
Comment (lisbet-alvarez) [reply addons/api/tests/unit/utils/mql-query-test.js:9]: this new structure makes it easier to read tests (esp for unit tests)

**Review by lisbet-alvarez — APPROVED** [review]
LGTM

**Pull request merged**